server/protocols/webadmin/webadmin-mailbox/src/test/java/org/apache/james/webadmin/routes/UserQuotaRoutesTest.java

```diff
@@ -0,0 +1,457 @@
+            given()
+                .get(QUOTA_USERS + "/" + BOB)
+            .then()
+                .statusCode(HttpStatus.OK_200)
+                .contentType(ContentType.JSON)
+                .extract()
+                .jsonPath();
+
+        assertThat(jsonPath.getLong(SIZE)).isEqualTo(maxStorage);
+        assertThat(jsonPath.getLong(COUNT)).isEqualTo(maxMessage);
+    }
+
+    @Test
+    public void getQuotaShouldReturnBothDefaultValues() throws Exception {
+        JsonPath jsonPath =
+            given()
+                .get(QUOTA_USERS + "/" + BOB)
+            .then()
+                .statusCode(HttpStatus.OK_200)
+                .contentType(ContentType.JSON)
+                .extract()
+                .jsonPath();
+
+        assertThat(jsonPath.getLong(SIZE)).isEqualTo(UNLIMITED);
+        assertThat(jsonPath.getLong(COUNT)).isEqualTo(UNLIMITED);
+    }
+
+    @Test
+    public void getQuotaShouldReturnBothWhenNoCount() throws Exception {
+        int maxStorage = 42;
+        maxQuotaManager.setMaxStorage(QuotaRoot.forUser(BOB), maxStorage);
+
+        JsonPath jsonPath =
+            given()
+                .get(QUOTA_USERS + "/" + BOB)
+            .then()
+                .statusCode(HttpStatus.OK_200)
+                .contentType(ContentType.JSON)
+                .extract()
+                .jsonPath();
+
+        assertThat(jsonPath.getLong(SIZE)).isEqualTo(maxStorage);
+        assertThat(jsonPath.getLong(COUNT)).isEqualTo(UNLIMITED);
+    }
+
+    @Test
+    public void getQuotaShouldReturnBothWhenNoSize() throws Exception {
+        int maxMessage = 42;
+        maxQuotaManager.setMaxMessage(QuotaRoot.forUser(BOB), maxMessage);
+
+
+        JsonPath jsonPath =
+            given()
+                .get(QUOTA_USERS + "/" + BOB)
+                .then()
+                .statusCode(HttpStatus.OK_200)
+                .contentType(ContentType.JSON)
+                .extract()
+                .jsonPath();
+
+        assertThat(jsonPath.getLong(SIZE)).isEqualTo(UNLIMITED);
+        assertThat(jsonPath.getLong(COUNT)).isEqualTo(maxMessage);
+    }
+
+    @Test
+    public void putQuotaShouldReturnNotFoundWhenUserDoesntExist() throws Exception {
+        when()
+            .put(QUOTA_USERS + "/" + JOE)
+        .then()
+            .statusCode(HttpStatus.NOT_FOUND_404);
+    }
+
+    @Test
+    public void putQuotaShouldUpdateBothQuota() throws Exception {
+        given()
+            .body("{\"" + COUNT + "\":52,\"" + SIZE + "\":42}")
+            .put(QUOTA_USERS + "/" + BOB)
+        .then()
+            .statusCode(HttpStatus.NO_CONTENT_204);
+
+        assertThat(maxQuotaManager.getMaxMessage(QuotaRoot.forUser(BOB))).isEqualTo(52);
+        assertThat(maxQuotaManager.getMaxStorage(QuotaRoot.forUser(BOB))).isEqualTo(42);
+    }
+
+    @Test
+    public void putQuotaShouldBeAbleToRemoveBothQuota() throws Exception {
+        given()
+            .body("{\"" + COUNT + "\":-1,\"" + SIZE + "\":-1}")
```

Comment:
what do you think of count: null to remove a quota?

Reply:
That's part of what I need to change : infinite is not the same as unset. but it requires quite deep refactoring.